### ISSUE TITLE
Key Generation

### ISSUE THREAD

**Issue by MarkLFT** [sjh37/efreversepoco#725]
Is there a way to configure the Key generation policy for the tables?  We plan to use HiLo rather than Identity.  We already have a Sequance defined that we use, but all of our configurations include `.ValueGeneratedOnAdd().UseIdentityColumn();` 

Where ideally we would prefer `.ValueGeneratedOnAdd().UseHiLi("sequencename","dbo");`  If that is not possible, then just `.ValueGeneratedOnAdd();` and we will use an partial class to set the global policy. 

Currently I am having to manually change the code on many table configurations to fix this.

Of have I missed a setting somewhere?

Thanks in advance.

**Comment by sjh37** [comment]
Hi Mark,

If you want to replace all of them, inside `EF.Reverse.POCO.v3.ttinclude`, approximately line 5850, change line:
```c#
databaseGeneratedOption += isEfCore3Plus ? ".UseIdentityColumn()" : ".UseSqlServerIdentityColumn()";
```
to
```c#
databaseGeneratedOption += isEfCore3Plus ? ".UseHiLo(\"sequencename\",\"dbo\")" : ".UseSqlServerIdentityColumn()";
```

**Comment by sjh37** [comment]
In near future, I'll make an enhacement to the generator which provides a callback to either use Identity or a HiLo sequence so you can specify if all tables or just a selected set of tables are to use HiLo sequences, and which sequence to use for each.

**Comment by sjh37** [comment]
Hi Mark, This work how now been completed and will be in the next release.

In the next release you can specify the following in your `<database>.tt` file
```cs
// Use the following list to add use of HiLo sequences for identity columns
Settings.HiLoSequences = new List<HiLoSequence>
{
    // Example
    // For column dbo.match_table_name.match_column_name this will use .UseHiLo("sequence_name", "sequence_schema") instead of .UseSqlServerIdentityColumn()
    new HiLoSequence
    {
        Schema         = "dbo",
        Table          = "match_table_name", // Use * to match all tables
        SequenceName   = "sequence_name",
        SequenceSchema = "sequence_schema"
    },
    new HiLoSequence
    {
        Schema         = "dbo",
        Table          = "Employees",
        SequenceName   = "EmployeeSequence",
        SequenceSchema = "dbo"
    }
};```

**Comment by MarkLFT** [comment]
Many thanks for this, it will save me a lot of work.  Greatly appreciated.

One thought/question, if I use a * for the table, I presume this will add the .UseHiLo() to each table configuration?  

Normally when hand-coding these classes, I use the .UseHiLi on the modelBuilder level, rather than in individual table configurations.  As I am using a single sequence for all tables, this makes the code a bit neater.   Like I said, just a thought for a future release. ;-)

**Comment by sjh37** [comment]
Now released in v3.5.0
